src/compiler/crystal/tools/init.cr

```diff
@@ -155,18 +180,36 @@ module Crystal
 
     class InitProject
       getter config : Config
+      getter views : Array(View)?
 
       def initialize(@config : Config)
       end
 
-      def run
-        views.each do |view|
-          view.new(config).render
+      def overwrite_checks
+        overwriting_views = views.select { |view| File.exists?(view.full_path) }
+        if overwriting_views.any?
+          if !config.force && !config.skip_existing
+            STDERR.puts "Cannot initialize Crystal project, the following files would be overwritten:"
+            overwriting_views.each do |view|
+              STDERR.puts "   #{"file".colorize(:red)} #{view.full_path} #{"already exist".colorize(:red)}"
+            end
+            STDERR.puts "You can use --force to overwrite those files,"
+            STDERR.puts "or --skip-existing to skip existing files and generate the others."
```

Comment:
How can I handle these messages properly using an exception?
I was thinking about using a custom exception with a custom field of the list of conflicting file paths, then in `compiler/crystal/command.cr` rescue this exception and print this long message using the custom field.

WDYT?

Reply:
So I implemented this, it looks ok, and works perfectly ;)